dev/NavigationView/NavigationView.cpp

```diff
@@ -3086,17 +3086,78 @@ void NavigationView::UpdateTopNavListViewItemSource(const winrt::IInspectable& i
 
 void NavigationView::UpdateListViewItemSource()
 {
-    if (!m_appliedTemplate)
-    {
-        return;
-    }
-
     auto dataSource = MenuItemsSource();
     if (!dataSource)
     {
         dataSource = MenuItems();
     }
 
+    auto iterable = dataSource.try_as<winrt::IIterable<winrt::IInspectable>>();
+    if (iterable)
+    {
+        // Certain items are disallowed in a NavigationView's items list. Check for them.
+        auto checkItemIsValid = [](const winrt::IInspectable& item)
```

Comment:
>checkItemIsValid [](start = 13, length = 16)

static checkItemIsValid?

Reply:
Why should I make it static? It's a lambda with no captures. This should be safe and cheap to pass around.